sdk/src/types/block/core/wrapper.rs

```diff
@@ -0,0 +1,348 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::mem::size_of;
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use packable::{
+    error::{UnexpectedEOF, UnpackError, UnpackErrorExt},
+    packer::{Packer, SlicePacker},
+    unpacker::{CounterUnpacker, SliceUnpacker, Unpacker},
+    Packable, PackableExt,
+};
+
+use crate::types::block::{
+    block_id::{BlockHash, BlockId},
+    core::{BasicBlock, ValidationBlock},
+    protocol::ProtocolParameters,
+    signature::Signature,
+    slot::{SlotCommitmentId, SlotIndex},
+    Block, Error, IssuerId,
+};
+
+/// Represent the object that nodes gossip around the network.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BlockWrapper {
+    /// Protocol parameters of the network to which this block belongs.
+    // TODO Not fully sure this is worth the small UX improvement, needs to be discussed more.
+    protocol_params: ProtocolParameters,
+    /// The time at which the block was issued. It is a Unix timestamp in nanoseconds.
+    issuing_time: u64,
+    /// The identifier of the slot to which this block commits.
+    slot_commitment_id: SlotCommitmentId,
+    /// The slot index of the latest finalized slot.
+    latest_finalized_slot: SlotIndex,
+    /// The identifier of the account that issued this block.
+    issuer_id: IssuerId,
+    /// The inner block.
+    block: Block,
```

Comment:
would have preferred this field to be name `inner` because that's what I kinda expect for a wrapper type :man_shrugging:

Reply:
We don't get to choose the names and order, it's the TIP. Changing the order would change the serialisation and ID, etc...